app/controllers/admin/api/services/proxies_controller.rb

```diff
@@ -63,13 +63,7 @@ def show
   ##~ op.parameters.add name: "jwt_claim_with_client_id_type", description: "JWT Claim With ClientId Type. Either `plain` or `liquid`", dataType: "string", paramType: "query", required: false
   #
   def update
-    if proxy.update_attributes(proxy_params)
-      if proxy.service_mesh_integration?
-        proxy.deploy!
-      elsif proxy.apicast_configuration_driven
-        proxy.deploy_v2
-      end
-    end
+    ProxyDeploymentService.new(proxy).deploy if proxy.update_attributes(proxy_params)
```

Comment:
Generally we use (by convention) `call` instead of `deploy` but that works too, no biggie :)